streampipes-client-python/streampipes_client/functions/broker/broker.py

```diff
@@ -0,0 +1,96 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from abc import ABC, abstractmethod
+from typing import AsyncIterator
+
+from streampipes_client.model.resource.data_stream import DataStream
+
+
+class Broker(ABC):
+    """Abstract implementation of a broker.
+    A broker is used to subscribe to a data stream and to consume the published events.
+    """
+
+    async def connect(self, data_stream: DataStream, host_address: str) -> None:
```

Comment:
What is the intention behind making the `connect` method asynchronous? 
Usually, you should wait until the connection was established successfully, right?
Or do I miss anything here?

Reply:
Yes, the method waits until the connection is established. The `async` is just needed because you have to use the `await` keyword to connect to the broker. This is because the `connect` method in the nats library is a coroutine.